src/test/java/org/mifos/integrationtest/cucumber/stepdef/BatchApiStepDef.java

```diff
@@ -363,7 +363,10 @@ public void iFetchBatchIDFromBatchTransactionAPISResponse() {
 
     @Then("I check for result file URL validity")
     public void iCheckForResultFileURLValidity() {
-        assertThat(isValidURL(scenarioScopeState.batchAndSubBatchSummaryResponse.getFile())).isTrue();
+        await().atMost(awaitMost, SECONDS).pollDelay(pollDelay, SECONDS).pollInterval(pollInterval, SECONDS).untilAsserted(() -> {
```

Comment:
Awaitly may not be needed here as, this block is not making any external call.
Are we waiting for any parallel process to fill data into this DTO? i dont think we have such a n architecture